### DISCUSSION TITLE
Custom Taffy struct

### DISCUSSION THREAD

**Discussion by DasLixou** [DioxusLabs/taffy#316] [Q&A] [answered]
Hey!
I wonder if it is possible to have a custom Taffy struct that stores the NodeData differently but still works the same and passes the same data to the algorythm? thanks :D

**Comment by nicoburns** [comment, marked as answer]
Hi! Yes, it is indeed possible. To do this you need to implement the `taffy::tree::LayoutTree` trait for your storage implementation and then call `taffy::compute_layout` on your type. This support is still quite new and hasn't been fully tested out, so please let us know if you have any trouble making this work and/or have any requested changes to Taffy to make this easier.

**Reply by DasLixou** [reply]
Thanks! I`ll try it :)

**Reply by DasLixou** [reply]
Hm.. i got this issue that `NodeData` is pub(crate) so i cant have a map with NodeData as a value. Any ideas?

**Reply by nicoburns** [reply]
`NodeData` is really an implementation details of Taffy's storage implementation (you'll note that it's not present in the `LayoutTree` trait). If you want to store data in the same way as `NodeData` does then you can just create your own struct that happens to have the same layout as `NodeData`. And of course you are then free to add additional fields to that struct too if you have need of them. Doing it like that allows for patterns like storing `Style`, `Layout` and `Cache` separately.

**Reply by DasLixou** [reply]
:o cool, thanks :D

**Reply by DasLixou** [reply]
It says that taffy::compute is a private module

**Reply by DasLixou** [reply]
nvm. just saw that taffy's lib.rs reexports it

**Reply by nicoburns** [reply]
Ah, sorry. As you might be able to tell, I haven't actually used this functionality myself yet (in fact, I think you might be the first person to use it!).
